### PR TITLE
Remove unused Index attributes

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -141,12 +141,10 @@ class Index(IndexOpsMixin, PandasObject):
     _join_precedence = 1
 
     # Cython methods
-    _arrmap = libalgos.arrmap_object
```
Comment (jreback) [inline pandas/core/indexes/base.py]: so I think you can also remove some of the cython code which implements (not float/object though)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/base.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexes/base.py]: arrmap_object is used in core.ops

Comment (jreback) [reply pandas/core/indexes/base.py]: right, but can remve the rest of the code. pls add to the list.

**Review by jreback — APPROVED** [review]

File: pandas/core/indexes/base.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexes/base.py]: > right, but can remve the rest of the code. pls add to the list.

The rest of what code?

Comment (jreback) [reply pandas/core/indexes/base.py]: arrmap in cython

**Pull request merged**